erpnext/erpnext_integrations/doctype/leaflink_settings/leaflink_settings.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+import json
+import pprint
+from frappe.utils import add_days
+import datetime
+import requests
+from frappe.model.document import Document
+
+class LeafLinkSettings(Document):
+	pass
+
+
+@frappe.whitelist()
+def sync_item_with_leaflink(item_details):
+	"""
+	Sync an item in leaflink by hitting the Sync with LeafLink Button.
+	"""
+	pass
+
+
+@frappe.whitelist(allow_guest=True)
```

Comment:
There should be an authentication for this endpoint. since its open to everyone.